crates/gitbutler-core/src/virtual_branches/integration.rs

```diff
@@ -314,9 +314,17 @@ fn verify_head_is_clean(
 
     let integration_commit = extra_commits.pop();
 
-    if integration_commit.is_none() {
-        // no integration commit found
-        return Err(errors::VerifyError::NoIntegrationCommit);
+    match &integration_commit {
+        Some(integration_commit) => {
+            if !is_integration_commit(integration_commit) {
+                // There's nothing to do, dont error out?
+                return Ok(());
+            }
+        }
+        None => {
+            // There's nothing to do, dont error out?
+            return Ok(());
+        }
```

Comment:
I think we.. might want to return an error here? If you run any command using `verify_branch` and we're not on the right head then it would feel unwise to let the command continue.